AnimationCodelab/finished/src/main/java/com/example/android/codelab/animation/ui/home/Home.kt

```diff
@@ -0,0 +1,702 @@
+            onClick = { onTabSelected(1) }
+        )
+    }
+}
+
+/**
+ * Shows an indicator for the tab.
+ *
+ * @param tabPositions The list of [TabPosition]s from a [TabRow].
+ * @param selectedTabIndex The index of the currently selected tab.
+ */
+@Composable
+private fun HomeTabIndicator(
+    tabPositions: List<TabPosition>,
+    selectedTabIndex: Int
+) {
+    val transition = updateTransition(selectedTabIndex)
+    val indicatorLeft by transition.animateDp(
+        transitionSpec = {
+            if (initialState < targetState) {
+                // Indicator moves to the right.
+                // Low stiffness spring for the left edge so it moves slower than the right edge.
+                spring(stiffness = Spring.StiffnessVeryLow)
+            } else {
+                // Indicator moves to the left.
+                // Medium stiffness spring for the left edge so it moves faster than the right edge.
+                spring(stiffness = Spring.StiffnessMedium)
+            }
+        }
+    ) { index ->
+        tabPositions[index].left
+    }
+    val indicatorRight by transition.animateDp(
+        transitionSpec = {
+            if (initialState < targetState) {
+                // Indicator moves to the right
+                // Medium stiffness spring for the right edge so it moves faster than the left edge.
+                spring(stiffness = Spring.StiffnessMedium)
+            } else {
+                // Indicator moves to the left.
+                // Low stiffness spring for the right edge so it moves slower than the left edge.
+                spring(stiffness = Spring.StiffnessVeryLow)
+            }
+        }
+    ) { index ->
+        tabPositions[index].right
+    }
+    val color by transition.animateColor { index ->
+        if (index == 0) purple700 else green800
+    }
+    Box(
+        Modifier
+            .fillMaxSize()
+            .wrapContentSize(align = Alignment.BottomStart)
+            .offset(x = indicatorLeft)
+            .width(indicatorRight - indicatorLeft)
+            .padding(4.dp)
+            .fillMaxSize()
+            .border(
+                BorderStroke(2.dp, color),
+                RoundedCornerShape(4.dp)
+            )
+    )
+}
+
+/**
+ * Shows a tab.
+ *
+ * @param icon The icon to be shown on this tab.
+ * @param title The title to be shown on this tab.
+ * @param onClick Called when this tab is clicked.
+ * @param modifier The [Modifier].
+ */
+@Composable
+private fun HomeTab(
+    icon: ImageVector,
+    contentDescription: String?,
+    title: String,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Row(
+        modifier = modifier
+            .clickable(onClick = onClick)
+            .padding(16.dp),
+        horizontalArrangement = Arrangement.Center,
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        Icon(
+            imageVector = icon,
+            contentDescription = contentDescription
```

Comment:
Again, is this CD needed as it will be merged with the text.

Reply:
Done.